### PR TITLE
feat: replace silent S3 fallbacks with explicit provider boundaries

### LINKED ISSUE #4411: Replace silent loader fallbacks with explicit provider boundaries and structured failure reporting
## What

Introduce explicit data-provider abstractions for local-filesystem and AWS-backed (S3) account/owner discovery in `backend/common/data_loader.py`, with typed result objects or domain exceptions distinguishing "missing data", "provider unavailable", and "invalid payload" — instead of converting all of these into empty results or silent local fallbacks.

## Why

`data_loader.py` currently mixes local filesystem discovery, S3 discovery, demo-owner logic, metadata parsing, virtual portfolios, and fallback behaviour in one module. Several provider failures are flattened into empty results without distinguishing transient outage from genuine "no data". Production incidents can therefore present as missing owners/accounts instead of actionable errors, which is hard to debug and risks serving incomplete data without an operator-visible signal.

## How

- Separate local and AWS account discovery/loading into distinct provider modules.
- Provider methods return typed models or raise well-defined exceptions instead of broad empty fallbacks.
- API routes distinguish missing data from provider outages and respond/log differently.
- Add tests covering S3 error, malformed JSON, empty payload, and fallback-to-local behaviour with explicit assertions.
- Preserve graceful degradation only where the API contract explicitly allows it, with structured logging/metrics on each fallback path.

## Constraints

- Existing callers of `data_loader` functions should keep working during the transition — consider a thin compatibility layer if needed rather than a single big-bang rename.
- Don't change the actual S3 bucket layout or local file format as part of this issue.

## LLM tier

**Sonnet** — requires introducing new abstractions and exception types across a module with several existing call sites; needs design judgment on the typed-model boundary.

## Success looks like

- Local and AWS discovery/loading live in separate provider modules.
- Distinct exception types exist for missing-data vs provider-unavailable vs invalid-payload.
- Tests explicitly cover all four failure-mode combinations described above.

## Failure looks like

- Fallback behaviour silently changes for existing working cases (e.g. demo owners stop resolving).
- Exceptions leak as unhandled 500s where the API previously returned a graceful empty response.

_From CODEBASE_REVIEW.md, item 2 ("v1 blockers")._

## Dependencies

| Relationship | Issue |
|---|---|
| Blocks | #4412 Introduce typed domain models |

### PR DESCRIPTION
**Pull request by leonarduk**
## Summary

- Removes the duplicate inline boto3 blocks from `load_account` and `load_person_meta` in `data_loader.py`, routing all S3 operations through `S3DataProvider` with explicit `MissingData` / `InvalidPayload` / `ProviderUnavailable` handling
- Fixes `data_providers._extract_person_meta` to not synthesise a `viewers: []` key when absent from the source document (the synthesised key was a latent inconsistency vs `data_loader._extract_person_meta`)
- Adds tests covering the four failure-mode combinations called out in the issue: S3 transport error, malformed JSON, empty payload, and fallback-to-local with structured warning log

## Test plan

- [ ] `pytest tests/test_data_loader_aws.py tests/test_data_loader_local.py` — 22 tests pass
- [ ] `ruff check backend/common/data_loader.py backend/common/data_providers.py` — no new errors (3 pre-existing in test file, unchanged)

Closes #4411

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved error handling and fallback mechanisms for account and person metadata loading when services are unavailable.
  * Enhanced validation of metadata to ensure data integrity and proper error reporting.
  * Refined account root configuration handling to provide clearer feedback when paths are misconfigured.

* **Tests**
  * Added comprehensive test coverage for metadata loading failure scenarios and error conditions.

### CODE REVIEW

File: backend/common/data_loader.py [hunk 1]
```diff
@@ -1108,68 +1078,35 @@ def _extract(data: Dict[str, Any]) -> Dict[str, Any]:
...
-                    sanitise_log_value(key),
-                    sanitise_log_value(exc),
-                    exc_info=True,
-                )
-                if not explicit_local_fallback:
-                    return {}
-        elif config.app_env == "aws" and not explicit_local_fallback:
-            # No bucket configured and no local fallback: nothing to load.
+        try:
+            return S3DataProvider().load_person_meta(owner).metadata
```
Comment (leonarduk) [inline backend/common/data_loader.py:1082]: why is so much code deleted?

**Review by leonarduk — COMMENTED** [review]
have we lost anything in this refactor?

**Comment by leonarduk** [comment]
Superseded by a clean branch — PR #4455 contained unrelated commits. Replacing with a single-commit PR from main.
